_posts/2020-06-23-Durr_Hoyer_Post_1.md

```diff
@@ -0,0 +1,328 @@
+    ---
+    operation ApplyConditionalPhase(register : LittleEndian) : Unit is Adj + Ctl 
+    {
+        using (aux = Qubit()) 
+        {
+            (ControlledOnInt(1,Z))(register!,aux); // If qubit is 1 flip it!
+        }
+    }
+
+---
+
+Our Q# script will be structured as follows:
+
+---
+
+    namespace QESA {
+        open Microsoft.Quantum.Intrinsic;
+        open Microsoft.Quantum.Diagnostics;
+        open Microsoft.Quantum.Arrays; 
+        open Microsoft.Quantum.Preparation;
+        open Microsoft.Quantum.Canon;
+        open Microsoft.Quantum.Arithmetic;
+        open Microsoft.Quantum.Math;
+        open Microsoft.Quantum.Core;
+        open Microsoft.Quantum.Convert;
+        operation Algorithm_Even(TableLength : Int, RandomIndex : Int) : Unit
+        {
+            using ((Register) = (Qubit[TableLength])) // intialize register to number of qubits as there are indices
+            {
+                within
+                    {   
+                        let Marker = Register[RandomIndex]; // Grab the qubit in the RandomIndex and set it aside
+
+                        PrepareUniformSuperposition(TableLength,LittleEndian(Register)); // Create Uniform Superposition of all indices
+
+                        Controlled Z(Register,Marker); // Apply Oracle to flip all states that are T[j]<T[y]
+                    }
+                apply 
+                    {
+                        ApplyToEach(H,Register); // Apply Hadamard to register
+
+                        ApplyConditionalPhase_0(LittleEndian(Register)); // Reflect qubits that are 0s
+
+                        ApplyToEachA(H,Register); // Apply Adjunct Hadamard to register
+
+                        ApplyConditionalPhase(LittleEndian(Register)); //Reflect qubits that are 1s
+                    }
+            }
+        }
+        
+        operation ApplyConditionalPhase_0(register: LittleEndian) : Unit is Adj + Ctl
+        {
+            using (aux = Qubit()) 
+            {
+                (ControlledOnInt(0,Z))(register!,aux); // If qubit is 0 flip it!
+            }
+        }
+        operation ApplyConditionalPhase(register : LittleEndian) : Unit is Adj + Ctl 
+        {
+            using (aux = Qubit()) 
+            {
+                (ControlledOnInt(1,Z))(register!,aux); // If qubit is 1 flip it!
+            }
+        }
+    }
+
+    
+---
+
+For futher information on how this was derived take a look at 'Tight bounds on quantum searching' [2].
+It is important to not the above algorithm only works for a table with even entries. 
+The algorithm breaks down when applying the Hadamard gate as the Hadamard is layed across the diagnoal of a identity matrix which is equal in dimensions to the number of qubits we have. 
+With a bit of math, if we try to lay a 2x2 matrix along an odd dimensioned identity matrix the transformation is not retained.
+To circumvent this we introduce the following implementation, utilizing QFT.

```

Comment:
To my understanding if we apply Hadamard to more than one qubit we lay it along the diagonal of the identity matrix, noting that Hadamard is a 2x2, if applied along a 3x3 identity matrix we lose information at the nth index of our diagonal because our 2x2 will fit in the first 2x2 but the 2x2 does not fit in the 1x1.